v2/pkg/ast/ast_enum_type_definition.go

```diff
@@ -56,6 +56,9 @@ func (d *Document) EnumTypeDefinitionHasEnumValueDefinition(ref int) bool {
 func (d *Document) EnumTypeDefinitionContainsEnumValue(enumTypeDef int, valueName ByteSlice) bool {
 	for _, i := range d.EnumTypeDefinitions[enumTypeDef].EnumValuesDefinition.Refs {
 		if bytes.Equal(valueName, d.EnumValueDefinitionNameBytes(i)) {
+			if _, isInaccessible := d.EnumValueDefinitionDirectiveByName(i, []byte("inaccessible")); isInaccessible {
```

Comment:
Would it make sense to reflect "inaccessible" in the AST directly?
You could then have 2 checks:
1. does the enum definition contain the value?
2. is the value accessible?
Currently, you return false for 1 when it's inaccessible when in reality it exists, but is not accessible, which, depending on the use case might not what the function caller expects.

Reply:
+1 this method should return every value which exists